### DISCUSSION TITLE
How can I add arguments on this Fork?

### DISCUSSION THREAD

**Discussion by leonnn1** [vladmandic/sdnext#138] [Q&A] [answered]
Hello, How to add arguments, on A1111 I could simply change the webui-user.bat, on this repo I dont know what to do!

**Comment by vladmandic** [comment]
- set environment variables by setting `COMMANDLINE_ARGS` in your system or user settings
- set environment variables by setting `COMMANDLINE_ARGS` in your shell
- edit shortcut to `webui.bat` and set command line params as properties there
- create any batch (e.g. `my-webui.bat`) with a single line: `webui.bat --whatever-param-you-want`

there are sooooo many standard ways. having prepackaged `webui-user.bat` is unnecessarily complicated and just silly.
and none of that is specific to this repo.

**Reply by leonnn1** [reply]
well, that is for users who are not programmers , A1111 was great because it allowed people who doesn't understand to be able to use it in a simple way for them to understand, so, for me all you said is way more complicated than a file with pre defined places to add stuff, again, you are treating users as programmers not regular people it seems

**Reply by PowerfulShell** [reply]
> * set environment variables by setting `COMMANDLINE_ARGS` in your system or user settings
> * set environment variables by setting `COMMANDLINE_ARGS` in your shell
> * edit shortcut to `webui.bat` and set command line params as properties there
> * create any batch (e.g. `my-webui.bat`) with a single line: `webui.bat --whatever-param-you-want`
> 
> there are sooooo many standard ways. having prepackaged `webui-user.bat` is unnecessarily complicated and just silly. and none of that is specific to this repo.

there are sooooo many standard ways. having prepackaged webui-user.bat is unnecessarily complicated and just silly.

i know right?

BUT why having prepackaged webui-user.bat is silly? 
since there are many people asking for where to put it? this makes me chuckled a bit  for illogical thinking of your statement.... 

isn't it more user friendly for starters to see what parameter they can use? and where to put them?

so many non-programmer use "webui" because they know nothing about CMD line......
otherwise what do you think "webui" exist in the first place?

if you subject everyone else are comfortable with cmd line, then you are dead wrong....

**Reply by vladmandic** [reply]
> edit shortcut to webui.bat and set command line params as properties there

it would take less time to do this than write the comment.
and what would be in pre-packaged `webui-user.bat`? prepackage it and leave it empty? what's the point?

**Comment by leonnn1** [comment]
Please, does anyone else that doesn't want to flex his coding skills but would be able to help a person who doesn't understand very much these things, me in this case?

**Reply by thinkaman** [reply]
To be blunt, the creator of this repo personally responded with multiple solutions, which is somewhat remarkable.  It's understandable if you needed additional guidance (and I'm pleased to see others have provided it), but channeling frustration at the person doing all of this for you (for free) is unproductive.  Nothing in his answer was "a flex", he was just offering multiple solutions to a vague question.

I only take the time to write this because the original Automatic1111 fork was overrun with thousands of bug reports + comments of less-technically-proficient users seeking basic personal troubleshooting in often rude attitudes, and I speculate that this culture of complaining played a part in the owner getting fed up and disappearing.

It's great that these leaps of technology are getting people to venture out into areas new to them, and I hope they continue--I don't want non-technical people excluded from the frontier.  But if you are going to insist on being on the _absolute cutting edge_ with no way of getting there yourself, some politeness to the dude carrying you is probably wise.

**Reply by leonnn1** [reply]
Chaneling grustration to a person who acts that he is superior to me because I don't know something? that acts as if he is saying, this is so dumb and simple and you don't know it? to be blunt, what I wrote was as fact to me as to him, if he wants to treat people like that he eventually is treated as he deserves.

**Comment by ke1ne** [comment, marked as answer]
@leonnn1 you can use this run.bat near the `automatic` dir, as a starting point:
```
@echo off
cd automatic
call webui.bat --autolaunch --no-download-sd-model --ckpt=your\full\path\to\models\v1-5-pruned-emaonly.safetensors --vae=your\full\path\to\vaes\vae-ft-mse-840000-ema-pruned.safetensors --upgrade
pause
```
To load faster, replace `--upgrade` to `--quick`. To see all flags available, run `webui.bat --help`.

**Reply by leonnn1** [reply]
Thank you I will try that now, sorry for the silly question, I am just trying to understand these things

**Reply by econundrum** [reply]
Or you could just create a shortcut to the webui.bat and add the arguments after the name as suggested it's quicker and easier than editing then creating another batch file.

**Reply by leonnn1** [reply]
THanks I understand a little better now, and I ended up doing what you said @econundrum I tried what @ke1ne explained and I messed up something, but in the end I got to this solution, thank you all anyways

**Comment by bdulfer** [comment]
Another way is to simply copy `webui-user.bat` from automatic1111 and modify it.
That way you have the same way to call the webui in both environments.

**Reply by leonnn1** [reply]
Thank you, this is what I ended up doing, it worked but I couldn't reach a solution to the problem why I needed to use arguments for, I need to learn more.

**Reply by lewisp95** [reply]
This worked perfectly for me, thanks for the tip @bdulfer
